pkg/cli/environment.go

```diff
@@ -124,7 +134,15 @@ func (s *EnvSettings) Namespace() string {
 //RESTClientGetter gets the kubeconfig from EnvSettings
 func (s *EnvSettings) RESTClientGetter() genericclioptions.RESTClientGetter {
 	s.configOnce.Do(func() {
-		s.config = kube.GetConfig(s.KubeConfig, s.KubeContext, s.namespace)
+		clientConfig := kube.GetConfig(s.KubeConfig, s.KubeContext, s.namespace)
+		if len(s.KubeToken) > 0 {
```

Comment:
generally speaking, `if s.KubeToken != ""` is the more idiomatic approach. This looks good though, and we can address it in a follow-up PR.

Reply:
https://github.com/helm/helm/pull/7721